### ISSUE TITLE
Change the docker base image for "otobo-web" to Debian:12-slim possible?

### ISSUE THREAD

**Issue by StefanRother-OTOBO** [RotherOSS/otobo-docker#142]
Hi,

We have created a test repository in github and dockerhub to be able to update the docker images in the future.

I noticed that by using the docker image ‘perl:5.38-bookworm’, we only close a few security gaps, because in my opinion there are too many layers in play, which then also have to be updated.

I therefore changed the repository ‘rotheross/otobodockerupdatetest’ to ‘debian:12-slim’ as a test and noticed that the image is much smaller and suddenly ‘green’.

<img width="970" alt="image" src="https://github.com/user-attachments/assets/36966f5a-10fe-45ac-975b-5deb7c91dc53">

We need to discuss this and consider whether I am making a mistake or whether it makes sense and would be possible to change this.

If it's about the specific Perl version, we could also include the part from the perl image in our Dockerfile.

Or am I missing something?

Thanks,

Stefan

**Comment by bschmalhofer** [comment]
It is of course possible to use the Debian slim base image. That is actually what the Perl image is using too. So the question is whether the Perl image pulls in a lot of stuff that OTOBO does not need. See the Dockerfile https://github.com/perl/docker-perl/blob/bc908b6f3523d1ce819ae2f26ddaabf11d955a4d/5.040.000-slim-bookworm/Dockerfile .

**Comment by StefanRother-OTOBO** [comment]
Yes, but I think the main advantage would be that we no longer have to wait for the updates of the intermediate layers.

If there is an update in the Debian image, this currently has no effect on our image, even if we regenerate it, but only when the ‘intermediate images’ have also been regenerated, or am I misunderstanding this?

**Comment by bschmalhofer** [comment]
Yes and no. Security fixes in Debian enter OTOBO only after the intermediate Perl image has been updated. But the Perl image is an Docker official image. My understanding is that these images are curated and security fixes are applied fast.

Checking the Perl and Debian images at https://hub.docker.com/layers/library/perl/5.40-slim-bookworm/images/sha256-9a1a7cbe079c58e16ba1bb3dcdd361effb416311dd1540066eeadf49c404a843?context=explore and https://hub.docker.com/layers/library/debian/bookworm-slim/images/sha256-d83056144b2dd301730d2739635c8cbdeaaae20d6887146434184f8c060f03ce?context=explore .
debian:bookworm-slim had been update 15 days ago and perl:5.40-slim-bookworm has been updated 14 days ago. I can't really tell whether that one day difference is significant. 

Looking at the layer hierarchy of the Perl image, I see that the most current Debian image is used indeed. I don't know details of the curation process for the Perl image. Apparently the updates in the Debian base image are considered. But I don't know whether security fixes in the installed Debian packages are checked as well, see https://github.com/perl/docker-perl/blob/bc908b6f3523d1ce819ae2f26ddaabf11d955a4d/5.040.000-slim-bookworm/Dockerfile.

**Comment by bschmalhofer** [comment]
Closing as we currently don't want to touch this.
